### PR TITLE
fixes bug 1311697 - deep copy SocorroDotDict better

### PR DESCRIPTION
**Pull request by peterbe**
I feel like I totally stressed through this. Lots of things going on today. 
This PR adds a test that does break unless you apply the fix. 
Also, this change does make the whole processor work. I've tested it by making the the crash gets sent in correctly (to my personal AWS S3 bucket). 

Let's get this to prod soon.

### CODE REVIEW

File: socorro/external/crashstorage_base.py [hunk 1]
```diff
@@ -603,7 +603,16 @@ def make_deep_copy(crash):
...
+            def dictify(thing):
+                for key, value in thing.items():
+                    if isinstance(value, SocorroDotDict):
+                        thing[key] = dict(value)
+                    if isinstance(value, collections.Mapping):
+                        dictify(value)
+
+            dictified = dict(crash)
+            dictify(dictified)
+            return SocorroDotDict(copy.deepcopy(dictified))
```
Comment (willkg) [inline socorro/external/crashstorage_base.py]: Why not extract dictify and then test it separately with unit tests?

**Comment by willkg** [comment]
I'll try fixing my issue here and pushing a commit.

**Comment by willkg** [comment]
^^^ I pushed stuff into your branch. That's the most amazing thing ever.

**Comment by willkg** [comment]
I have no idea what we do with reviews that have two authors. Do we need a third separate pair of eyes? Can we review each others' stuff? Does @lonnen have to come down from his throne of skulls to bless us with new powers?

**Comment by adngdb** [comment]
@willkg the ``dictify`` function can be removed now, and replaced with a call to ``deepcopy``. Do you want to do that? I can push to Peter's branch as well, and then this PRs becomes like a big Socorro party!

**Comment by willkg** [comment]
@adngdb I don't understand what you mean by that. How can `dictify` or the new `socorrodotdict_to_dict` be replaced with a deepcopy?

**Comment by willkg** [comment]
@adngdb Oh, my bad--I didn't finish. My sincerest apologies. I'll fix it now. Bad naughty me.

**Comment by willkg** [comment]
@adngdb I think that fixes it. The tests pass locally. If you see any issues, you should push a commit, too. Then we can all spend tomorrow figuring out what the new process should be.

🎉 🎈 🍰

**Comment by peterbe** [comment]
I'll merge this once it works running as my local processor.

**Pull request merged**